webapp/utils/aws/ecs.py

```diff
@@ -0,0 +1,118 @@
+import time
+from typing import TYPE_CHECKING, Literal
+
+import boto3
+from attrs import define
+
+if TYPE_CHECKING:
+    from mypy_boto3_ecs.client import ECSClient as ECSClientType
+
+from webapp.exceptions import ECSTaskDefinitionDoesNotExistError
+
+logger = logging.getLogger(__name__)
+
+
+@define
+class ECSClient:
+    """ECS Client for running and monitoring task runs."""
+
+    cluster: str
+    task_definition: str
+    network_configuration: dict
+    container: str
+
+    @property
+    def client(self) -> "ECSClientType":
+        return boto3.client("ecs")
+
+    @property
+    def task_family(self) -> str:
+        task_family_revision = self.task_definition.split("/")[-1]
+        return re.sub(":.*", "", task_family_revision)
+
+    def execute_review_run(self) -> dict | None:
+        return self.run(run_type="review")
+
+    def execute_final_run(self) -> dict | None:
+        return self.run(run_type="final")
+
+    def run(self, run_type: Literal["review", "final"]) -> dict | None:
+        if run_type == "review":
+            logger.info("Executing ECS task for 'Review Run'")
+            commands = ["--real"]
+        elif run_type == "final":
+            logger.info("Executing ECS task for 'Final Run'")
+            commands = ["--real", "--final"]
+        else:
+            message = f"Cannot run task for unrecognized run_type='{run_type}'"
+            raise RuntimeError(message)
+
+        if self.task_exists():
+            response = self.client.run_task(
+                cluster=self.cluster,
+                launchType="FARGATE",
+                networkConfiguration=self.network_configuration,  # type: ignore[arg-type]
+                overrides={
+                    "containerOverrides": [
+                        {
+                            "name": self.container,
+                            "command": commands,
+                        }
+                    ]
+                },
+                taskDefinition=self.task_definition,
+            )
+            return self.parse_run_details(response)  # type: ignore[arg-type]
+        return None
+
+    def monitor_task(self, task: str) -> dict:
+        """Polls ECS for task status updates.
+
+        This method will periodically (every 5 seconds) check for the status
+        of the running ECS task that matches the provided task ARN.
+        When the task run is complete, details of the run are returned.
+        """
+        while True:
+            response = self.client.describe_tasks(cluster=self.cluster, tasks=[task])
+            task_run_details = self.parse_run_details(response)  # type: ignore[arg-type]
+            task_status = task_run_details["status"]
+
+            message = f"Status for task {task}: {task_status}"
+            logger.info(message)
+
+            if task_status == "STOPPED":
+                logger.info("Task run has completed.")
+                break
+            time.sleep(5)
+        return task_run_details
```

Comment:
This function will likely be updated once we have a clearer idea of how to display running tasks on the front end (i.e., loading / status updates). For now, this function demonstrates how `boto3.ECS.Client.describe_task` can be used to check on the status of a running ECS task.